objects/src/accounts/storage/map.rs

```diff
@@ -0,0 +1,98 @@
+use super::{AccountError, Word};
+use crate::crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{InnerNodeInfo, LeafIndex, Smt, SmtLeaf, SmtProof, SMT_DEPTH},
+};
+
+// ACCOUNT STORAGE MAP
+// ================================================================================================
+
+/// Account storage map is a Sparse Merkle Trees of depth 64. It can be used to store more data as
+/// there is in plain usage of the storage slots. The root of the SMT consumes one account storage
+/// slot.
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct StorageMap {
+    map: Smt,
+}
+
+impl StorageMap {
```

Comment:
Question: Do we need this struct? Why not use the `Smt` directly?

Reply:
Excellent question. I was also still determining. 

One reason to introduce `StorageMap` instead of using `Smt` directly is that now I am able to define a custom serialization / deserialization for `StorageMap` which could be different from `Smt`. 

@bobbinth  had an opinion about it. I can implement both.

Reply:
I think it is a bit cleaner to have a newtype for storage maps as we can add custom functionality to it. For now, this may be just serializaiton/deserialization, but in the future we'll probably add more (to handle different value arities, for example).